### DISCUSSION TITLE
Understanding JSX pragma

### DISCUSSION THREAD

**Discussion by iamskok** [system-ui/theme-ui#1620] [Q&A] [answered]
What is the difference between `/** @jsxImportSource theme-ui */` and `/** @jsx jsx */`? When should I use one over another?

I saw both of the instances being used in the documentation [sx-prop](https://theme-ui.com/sx-prop/).
`

**Comment by mrmartineau** [comment, marked as answer]
@iamskok they are actually explained in more detail here: https://theme-ui.com/api#jsx-runtime

Top line: use `/** @jsxImportSource theme-ui */` if you use React's automatic JSX runtime, as explained in this [blog post](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html), otherwise use `/** @jsx jsx */`
